wallet-api/src/Ledger/Map/TH.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE UndecidableInstances   #-}
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE DeriveAnyClass         #-}
+{-# LANGUAGE DeriveGeneric          #-}
+{-# LANGUAGE DerivingStrategies     #-}
+{-# LANGUAGE TemplateHaskell        #-}
+{-# LANGUAGE LambdaCase             #-}
+{-# LANGUAGE MonoLocalBinds         #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+-- A map implementation that can be used in on-chain and off-chain code.
+module Ledger.Map.TH(

```

Comment:
We probably want a non-TH version of this, people may well want to use it in other places too.